sgt-back-end/index.js

```diff
@@ -0,0 +1,132 @@
+const express = require('express');
+const pg = require('pg');
+const app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('listening on port 3000!');
+});
+
+// GET request to return all rows from the "grades" table
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades";
+  `;
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+});
+
+// POST request to insert a new grade into the "grades" table
+app.post('/api/grades', (req, res, next) => {
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values($1, $2, $3)
+    returning *;
+  `;
+  const values = [req.body.name, req.body.course, req.body.score];
+  // checks if required fields are completed and are valid entries
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    res.status(400).json({ error: 'missing required name, course, or grade field' });
+  } else if (typeof values[0] !== 'string' || typeof values[1] !== 'string') {
+    res.status(400).json({ error: 'the name and course must be a valid alphabetical characters' });
+  } else if ((typeof values[2] !== 'number') && values[2] < 0) {
+    res.status(400).json({ error: 'the grade must be a positive number' });
```

Comment:
Please make sure you're correctly validating the score's range:

---

![image](https://user-images.githubusercontent.com/16143771/201264617-d71d6590-ecfa-4e67-bab6-852aa7022eba.png)

Reply:
<img width="538" alt="Screen Shot 2022-11-10 at 9 02 18 PM" src="https://user-images.githubusercontent.com/103090771/201267151-fb89a42f-46e4-4c3b-8130-74ab13d93b6a.png">